middleware/ft800/lib/src/ft800.c

```diff
@@ -0,0 +1,1913 @@
+/****************************************************************************
+**
+** Copyright (C) ${COPYRIGHT_YEAR} MikroElektronika d.o.o.
+** Contact: https://www.mikroe.com/contact
+**
+** This file is part of the mikroSDK package
+**
+** Commercial License Usage
+**
+** Licensees holding valid commercial NECTO compilers AI licenses may use this
+** file in accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and The MikroElektronika Company.
+** For licensing terms and conditions see
+** https://www.mikroe.com/legal/software-license-agreement.
+** For further information use the contact form at
+** https://www.mikroe.com/contact.
+**
+**
+** GNU Lesser General Public License Usage
+**
+** Alternatively, this file may be used for
+** non-commercial projects under the terms of the GNU Lesser
+** General Public License version 3 as published by the Free Software
+** Foundation: https://www.gnu.org/licenses/lgpl-3.0.html.
+**
+** The above copyright notice and this permission notice shall be
+** included in all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** OF MERCHANTABILITY, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+** TO THE WARRANTIES FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+** DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT
+** OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
+** OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+**
+****************************************************************************/
+/*!
+ * @file  ft800.c
+ * @brief FT800 controller source file.
+ */
+
+#include "ft800.h"
+#include "ft800_cmd.h"
+#include "ft800_reg.h"
+#include <stdint.h>
+#include <math.h>
+
+const ft800_controller_t FT800_CONTROLLER =
+{
+    {
+        { 0x00, TP_EVENT_GEST_NONE },
+        { 0x1C, TP_EVENT_GEST_LEFT },
+        { 0x14, TP_EVENT_GEST_RIGHT },
+        { 0x10, TP_EVENT_GEST_UP },
+        { 0x18, TP_EVENT_GEST_DOWN },
+        { 0x48, TP_EVENT_GEST_ZOOM_IN },
+        { 0x49, TP_EVENT_GEST_ZOOM_OUT }
+    }
+};
+
+void ft800_cfg_setup( ft800_cfg_t *cfg, const ft800_controller_t *controller )
+{
+    spi_master_select_device( cfg->cs_pin );
+
+    cfg->in_pin = HAL_PIN_NC;
+    cfg->controller = controller;
+}
+
+void write_data( ft800_t *ctx,ft800_cfg_t *cfg, uint32_t addres, uint32_t value, uint8_t length )
+{
+    uint8_t data[ 7 ];
+    
+    spi_master_select_device( cfg->cs_pin );
+    
+    if ( length == 24 )
```

Comment:
At MikroE we always place constant value on the left size in if conditions. This is a good practice not to leave a single `=` sign and make an error. And not to forget - this random value should be a macro. So in this case it should look like:
```suggestion
    if ( FT800_DATA_LENGTH_BYTES_3 == length )
```